feat(cli): claim from snapshot using get-faucet

iancoleman opened the pull request:
This introduces a second way to get tokens from the faucet using the cli

`safe wallet get-faucet <url>`
Retains the existing fixed amount of tokens into the current wallet.
    
`safe wallet get-faucet <url> <maid_address> <signature>`
Receives a transfer from the maid snapshot for the amount of maid owned
by that address. The signature is a bitcoin signature by the maid
address with the message as the hex encoded safe wallet address that
will receive the transfer.

This uses the current safe wallet as the recipient.

It also introduces a few helpers for wallet management, not currently used by the cli but will be handy to include in future commands `safe wallet stash` and `safe wallet unstash <address>`

* `stash` - moves the current wallet into a new directory `wallet_<address>`
* `unstash` - moves the stashed wallet for the given address to the `wallet` directory - need to double check behavior for overwriting the existing wallet here (or not).
* `remove` - deletes the current wallet stored in `wallet` directory

These helpers arose for two main reasons

* there's no way to create an in-memory only wallet, mainly because of the dependency on `cash_notes` and `payments` files to make the wallet work correctly, so it's very common to move wallets around
* the backup for most blockchain systems is just the keys (usually via mnemonics) but this isn't enough to create a backup of a safe wallet, so these helpers are a starting point for backing up wallets.

## Description

reviewpad:summary

davidrusu commented on sn_node/src/node/flow_ctrl/mod.rs:
```diff
@@ -276,35 +276,33 @@ impl FlowCtrl {
-                                wire_msg.msg_id()
-                            );
+                        let name = recipients[0].name();
+                        let dst = sn_interface::messaging::DstLocation::Node {
```
Is Node the correct destination here? Would `DstLocation::Section` fit better?

joshuef replied:
No, it's only going to the specific nodes that need this data.

oetyng commented:
fat fingered the rerun of jobs.. was supposed to rerun failed

iancoleman committed:
feat(cli): get-faucet has maid address option

This introduces a second way to get tokens from the faucet using the cli

`safe wallet get-faucet <url>`
Retains the existing fixed amount of tokens into the current wallet.

`safe wallet get-faucet <url> <maid_address> <signature>
Receives a transfer from the maid snapshot for the amount of maid owned
by that address. The signature is a bitcoin signature by the maid
address with the message as the hex encoded safe wallet address that
will receive the transfer.

joshuef commented on sn_cli/src/subcommands/wallet/helpers.rs:
```diff
@@ -6,13 +6,48 @@
+    address: Option<String>,
+    signature: Option<String>,
+) -> Result<()> {
+    if address.is_some() && signature.is_some() {
```
We want to avoid `expect` in there, I know the outer check should make that fine, but if we were refactoring down the line this could open a can of `panic` worms instead of graceful error handling.

It's preferred to 
```rust
let addr = if let Some(addr) = address {
addr } 
else {
return Error(<SomeClearError> };

```
and handle that case explicitly with an error we can handle here/ elsewhere, rather than potentially panicking.

same for `sig`

iancoleman replied:
Thanks for the tip, always nice to get advice that removes potential panics. Fixed in https://github.com/maidsafe/safe_network/commit/34ea9e872e76ecce64ba7ba82492524a0a49a5ff

JasonPaulGithub committed:
Merge branch 'alpha' into faucet_cli
